lib/metasploit/framework/obfuscation/crandomizer/modifier.rb

```diff
@@ -0,0 +1,182 @@
+require 'metasploit/framework/obfuscation/crandomizer/random_statements'
+
+module Metasploit
+  module Framework
+    module Obfuscation
+      module CRandomizer
+
+        class Modifier
+          attr_reader :parser
+          attr_reader :fake_functions
+          attr_reader :weight
+
+          # Initializes a Metasploit::Framework::Obfuscation::CRandomizer::Modifier instance.
+          #
+          # @param p [Metasploit::C::Parser]
+          # @param f [Metasploit::Framework::Obfuscation::CRandomizer::CodeFactory::FakeFunctionCollection]
+          # @param w [Integer] Weight of the randomness.
+          def initialize(p, f, w)
+            @parser = p
+            @fake_functions = f
+            @weight = w
+          end
+
+          # Modifies different if-else blocks recursively.
+          #
+          # @param s [Metasm::C::Declaration]
+          # @return [Metasm::C::Declaration]
+          def modify_if_else_blocks(s)
+            modify_if(s)
+            modify_else_if(s)
+            modify_else(s)
+            s
+          end
+
+          # Modifies an if block.
+          #
+          # @param s [Metasm::C::Declaration]
+          # return [void]
+          def modify_if(s)
+            new_if_statements = []
+
+            s.bthen.statements.each do |stmt|
+              modify_nested_blocks(stmt)
+              new_if_statements.concat(get_fake_statement)
+              new_if_statements << stmt
+            end
+
+            s.bthen.statements = new_if_statements
+          end
+
+          # Modifies an else-if block.
+          #
+          # @param s [Metasm::C::Declaration]
+          # @param [void]
+          def modify_else_if(s)
+            # There could be multiple else if blocks,
+            # this gives the current else if block
+            elseif_block = s.belse
+
+            while (elseif_block && elseif_block.respond_to?(:bthen)) do
+              new_else_if_statements = []
+
+              elseif_block.bthen.statements.each do |stmt|
+                modify_nested_blocks(stmt)
+                new_else_if_statements.concat(get_fake_statement)
+                new_else_if_statements << stmt
+              end
+
+              elseif_block.bthen.statements = new_else_if_statements
+
+              # Move on to the next else if block
+              elseif_block = elseif_block.belse
+            end
+          end
+
+          # Modifies an else block.
+          #
+          # @param s [Metasm::C::Declaration]
+          def modify_else(s)
+            else_block = s.belse
+
+            # The else block is retrieved this way when there is an else if block
+            else_block = s.belse.belse if s.belse.respond_to?(:belse)
+
+            # There is really no else block, let's bail.
+            # return unless else_block
+            return unless else_block.respond_to?(:statements)
+
+            new_else_statements = []
+
+            else_block.statements.each do |stmt|
```

Comment:
This block of code appears in multiple place.
Maybe put the statements in a function and call it.

Reply:
Yeah I've thought about that, and then it was actually more difficult to read so I ended up giving up on that idea. If it's really better to do that way, let me know and I can try again. Thanks.